Description: modernize use of NodeJS module acorn
Origin: upstream, https://github.com/eslint/espree/commit/ee1d3ec
Bug: https://github.com/eslint/espree/issues/514
Bug-Debian: https://bugs.debian.org/997789
Author: Milos Djermanovic <milos.djermanovic@gmail.com>
Last-Update: 2021-11-07
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
--- a/package.json
+++ b/package.json
@@ -18,7 +18,7 @@
   },
   "license": "BSD-2-Clause",
   "dependencies": {
-    "acorn": "^7.4.0",
+    "acorn": "^8.5.0",
     "acorn-jsx": "^5.3.1",
     "eslint-visitor-keys": "^1.3.0"
   },
--- a/tests/fixtures/ecma-version/10/json-superset/valid-2028.result.js
+++ b/tests/fixtures/ecma-version/10/json-superset/valid-2028.result.js
@@ -6,8 +6,8 @@
             "column": 0
         },
         "end": {
-            "line": 1,
-            "column": 11
+            "line": 2,
+            "column": 1
         }
     },
     "range": [
@@ -23,8 +23,8 @@
                     "column": 0
                 },
                 "end": {
-                    "line": 1,
-                    "column": 11
+                    "line": 2,
+                    "column": 1
                 }
             },
             "range": [
@@ -40,8 +40,8 @@
                             "column": 4
                         },
                         "end": {
-                            "line": 1,
-                            "column": 11
+                            "line": 2,
+                            "column": 1
                         }
                     },
                     "range": [
@@ -74,8 +74,8 @@
                                 "column": 8
                             },
                             "end": {
-                                "line": 1,
-                                "column": 11
+                                "line": 2,
+                                "column": 1
                             }
                         },
                         "range": [
@@ -155,8 +155,8 @@
                     "column": 8
                 },
                 "end": {
-                    "line": 1,
-                    "column": 11
+                    "line": 2,
+                    "column": 1
                 }
             },
             "range": [
--- a/tests/fixtures/ecma-version/10/json-superset/valid-2029.result.js
+++ b/tests/fixtures/ecma-version/10/json-superset/valid-2029.result.js
@@ -6,8 +6,8 @@
             "column": 0
         },
         "end": {
-            "line": 1,
-            "column": 11
+            "line": 2,
+            "column": 1
         }
     },
     "range": [
@@ -23,8 +23,8 @@
                     "column": 0
                 },
                 "end": {
-                    "line": 1,
-                    "column": 11
+                    "line": 2,
+                    "column": 1
                 }
             },
             "range": [
@@ -40,8 +40,8 @@
                             "column": 4
                         },
                         "end": {
-                            "line": 1,
-                            "column": 11
+                            "line": 2,
+                            "column": 1
                         }
                     },
                     "range": [
@@ -74,8 +74,8 @@
                                 "column": 8
                             },
                             "end": {
-                                "line": 1,
-                                "column": 11
+                                "line": 2,
+                                "column": 1
                             }
                         },
                         "range": [
@@ -155,8 +155,8 @@
                     "column": 8
                 },
                 "end": {
-                    "line": 1,
-                    "column": 11
+                    "line": 2,
+                    "column": 1
                 }
             },
             "range": [
--- a/tests/fixtures/ecma-version/6/newTarget/invalid-new-target.result.js
+++ b/tests/fixtures/ecma-version/6/newTarget/invalid-new-target.result.js
@@ -2,5 +2,5 @@
     "index": 8,
     "lineNumber": 1,
     "column": 9,
-    "message": "'new.target' can only be used in functions"
+    "message": "'new.target' can only be used in functions and class static block"
 };
